assets/javascripts/discourse/initializers/chat-setup.js

```diff
@@ -129,6 +131,28 @@ export default {
           }
         });
       });
+
+      api.modifyClass("model:user", {
+        pluginId: "discourse-chat",
+
+        @discourseComputed("staff", "groups.[]")
+        allowDirectMessages() {
+          return (
+            this.staff ||
+            this.isInAnyGroups(
+              this.siteSettings.direct_message_enabled_groups
+                .split("|")
+                .map((groupId) => parseInt(groupId, 10))
+            )
+          );
+        },
+      });
+
+      // TODO (martin) Better way to do this??
```

Comment:
@jjaffeux I wanted to ask for your advice on this, unless I either did this or, mysteriously, called `User.create()` after modifying the class, I was unable to call `allowDirectMessages` on the current user. Am I doing something wrong or is this fine to do?